tests/Dockerfile.daemon

```diff
@@ -5,29 +5,30 @@ FROM ubuntu:20.04 AS build
 ENV DEBIAN_FRONTEND=noninteractive
 RUN apt update && apt install -y make git build-essential golang liblzma-dev jq libssl-dev libglib2.0-dev
 
-COPY ./ /go/src/github.com/mendersoftware/mender/
-RUN make -C /go/src/github.com/mendersoftware/mender GOPATH=/go clean
-RUN make -C /go/src/github.com/mendersoftware/mender GOPATH=/go prefix=/mender-install install
-RUN sh -c 'jq ".ServerCertificate=\"/usr/share/doc/mender-client/examples/demo.crt\" | .ServerURL=\"https://docker.mender.io/\"" \
-    < /go/src/github.com/mendersoftware/mender/examples/mender.conf.demo > /mender-install/etc/mender/mender.conf'
+WORKDIR /go/src/github.com/mendersoftware/mender
+COPY ./ .
+RUN make GOPATH=/go clean
+RUN make GOPATH=/go prefix=/mender-install install
+RUN jq ".ServerCertificate=\"/usr/share/doc/mender-client/examples/demo.crt\" | .ServerURL=\"https://docker.mender.io/\"" \
+    < examples/mender.conf.demo > /mender-install/etc/mender/mender.conf
+RUN echo artifact_name=original > /mender-install/etc/mender/artifact_info
 
-FROM ubuntu:18.04
+FROM ubuntu:20.04
 
 RUN apt update && apt install -y openssh-server
 
 # Set no password
 RUN sed -ie 's/^root:[^:]*:/root::/' /etc/shadow
 RUN sed -ie 's/^UsePAM/#UsePam/' /etc/ssh/sshd_config
-RUN sh -c 'echo PermitEmptyPasswords yes >> /etc/ssh/sshd_config'
-RUN sh -c 'echo PermitRootLogin yes >> /etc/ssh/sshd_config'
-
-RUN sh -c 'echo Port 22 >> /etc/ssh/sshd_config'
-RUN sh -c 'echo Port 8822 >> /etc/ssh/sshd_config'
-
-COPY --from=build /mender-install/ /
-
-RUN sh -c 'mkdir -p /var/lib/mender && echo device_type=docker-client > /var/lib/mender/device_type'
-RUN sh -c 'echo artifact_name=original > /etc/mender/artifact_info'
+RUN echo 'PermitEmptyPasswords yes\n\
+PermitRootLogin yes\n\
+Port 22\n\
+Port 8822\n' >> /etc/ssh/sshd_config
```

Comment:
Are you sure that this works? `echo` does not translate `\n` sequences, so you end up with a literal backslash and 'n'. That is, unless Docker translates it.

Reply:
This part actually works - I just verified that the lines as correctly appended to `sshd_config`. But something else is broken: the three integration tests that use this image are all failing :)

Investigating...